src/PlusVolumeReconstruction/Testing/CMakeLists.txt

```diff
@@ -1,104 +1,33 @@
 SET( TestDataDir ${PLUSLIB_DATA_DIR}/TestImages )
 SET( ConfigFilesDir ${PLUSLIB_DATA_DIR}/ConfigFiles )
 
-IF(PLUSBUILD_BUILD_PlusLib_TOOLS)
-  # Reconstruct volumes
-  ADD_TEST(vtkVolumeReconstructorTestRunNearLateUChar
-    ${PLUS_EXECUTABLE_OUTPUT_PATH}/VolumeReconstructor
-    --config-file=${ConfigFilesDir}/Testing/PlusDeviceSet_VolumeReconstructionOnly_SonixRP_TRUS_D70mm_NN_LATE.xml
-    --source-seq-file=${TestDataDir}/SpinePhantomFreehand.mha
-    --output-volume-file=vtkVolumeReconstructorTestNNLATEvolume.mha
-    --image-to-reference-transform=ImageToReference
-    )
-  SET_TESTS_PROPERTIES( vtkVolumeReconstructorTestRunNearLateUChar PROPERTIES FAIL_REGULAR_EXPRESSION "ERROR;WARNING" )
-
-  ADD_TEST(vtkVolumeReconstructorTestRunNearMeanUChar
-    ${PLUS_EXECUTABLE_OUTPUT_PATH}/VolumeReconstructor
-    --config-file=${ConfigFilesDir}/Testing/PlusDeviceSet_VolumeReconstructionOnly_SpinePhantom_NN_MEAN.xml
-    --source-seq-file=${TestDataDir}/SpinePhantomFreehand.mha
-    --output-volume-file=vtkVolumeReconstructorTestNNMEANvolume.mha
-    --image-to-reference-transform=ImageToReference
-    )
-  SET_TESTS_PROPERTIES( vtkVolumeReconstructorTestRunNearMeanUChar PROPERTIES FAIL_REGULAR_EXPRESSION "ERROR;WARNING" )
-
-  ADD_TEST(vtkVolumeReconstructorTestRunNearMaxiFloat
-    ${PLUS_EXECUTABLE_OUTPUT_PATH}/VolumeReconstructor
-    --config-file=${ConfigFilesDir}/Testing/PlusDeviceSet_VolumeReconstructionOnly_SpinePhantom_NN_MAXI.xml
-    --source-seq-file=${TestDataDir}/SpinePhantomFreehand3FramesFloat.mha
-    --output-volume-file=vtkVolumeReconstructorTestNNMAXIvolume.mha
-    --image-to-reference-transform=ImageToReference
-    )
-  SET_TESTS_PROPERTIES( vtkVolumeReconstructorTestRunNearMaxiFloat PROPERTIES FAIL_REGULAR_EXPRESSION "ERROR;WARNING" )
-
-  ADD_TEST(vtkVolumeReconstructorTestRunLinrLateFloat
-    ${PLUS_EXECUTABLE_OUTPUT_PATH}/VolumeReconstructor
-    --config-file=${ConfigFilesDir}/Testing/PlusDeviceSet_VolumeReconstructionOnly_SpinePhantom_LN_LATE.xml
-    --source-seq-file=${TestDataDir}/SpinePhantomFreehand3FramesFloat.mha
-    --output-volume-file=vtkVolumeReconstructorTestLNLATEvolume.mha
-    --image-to-reference-transform=ImageToReference
-    )
-  SET_TESTS_PROPERTIES( vtkVolumeReconstructorTestRunLinrLateFloat PROPERTIES FAIL_REGULAR_EXPRESSION "ERROR;WARNING" )
-
-  ADD_TEST(vtkVolumeReconstructorTestRunLinrMeanUChar
-    ${PLUS_EXECUTABLE_OUTPUT_PATH}/VolumeReconstructor
-    --config-file=${ConfigFilesDir}/Testing/PlusDeviceSet_VolumeReconstructionOnly_SonixRP_TRUS_D70mm_LN_MEAN.xml
-    --source-seq-file=${TestDataDir}/SpinePhantomFreehand.mha
-    --output-volume-file=vtkVolumeReconstructorTestLNMEANvolume.mha
-    --image-to-reference-transform=ImageToReference
-    )
-  SET_TESTS_PROPERTIES( vtkVolumeReconstructorTestRunLinrMeanUChar PROPERTIES FAIL_REGULAR_EXPRESSION "ERROR;WARNING" )
-
-  ADD_TEST(vtkVolumeReconstructorTestRunLinrMaxiUChar
+function(RegressionTest TestName Config Seq Out)

```

Comment:
That was not just hard to look at, but also not very conductive of uniformity :smile: